tvm/src/schedule/schedule_reorder.cc

```diff
@@ -16,163 +16,227 @@
 namespace TVM {
 namespace schedule {
 
+bool debug = true;
+
 using namespace ir;
-bool debug = false;
 
-// TODO(Hecmay): construct sch->stage_buf_map_
-// Update map from stage to its parent stages
+using std::string;
+using std::unordered_map;
+using std::unordered_set;
+using std::vector;
+
+// Store the stage placement information
+struct DevScope {
+  DeviceType dev_type;
+  bool is_endpoint{false};
+  StorageType storage_type;
+  StreamType stream_type;
+  int mem_port{-1};
+  int channel_depth{-1};
+  int burst_len{-1};
+  string target_tensor;
+};
+
+// (Top level) Basic Blocks in stage body. The types can be
+// 1. AtttStmt attachments
+// 2. Nested for loops
+struct BasicBlock {
```

Comment:
We really shouldn't name it BasicBlock when it has another more common definition. We can name it ElemBlock or BlockElem.